### ISSUE TITLE
Gentoo package

### ISSUE THREAD

**Issue by kobzar** [minbrowser/min#1267]
Could you provide package for Gentoo Linux?

**Comment by BlueManCZ** [comment]
I maintain an [.ebuild](https://github.com/BlueManCZ/edgets/tree/master/www-client/min-bin) for min browser in [Edgets](https://github.com/BlueManCZ/edgets) overlay, so you can install it from there.

Just:
```
sudo layman --add edgets
sudo emerge --ask min-bin
```

**Comment by genomorro** [comment]
Hi.

I just tried this ebuild code, unfortunately `npm install` doesn't work, it never downloads dependencies. Manually, this is working. If anyone can help me fix it, I would appreciate it.

```
# Copyright 1999-2022 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=8
inherit unpacker xdg

DESCRIPTION="Min is a fast, minimal browser that protects your privacy."
HOMEPAGE="https://minbrowser.github.io/min/"
SRC_URI="https://github.com/minbrowser/${PN}/archive/refs/tags/v${PV}.tar.gz"

LICENSE="Apache-2.0"
SLOT="0"
KEYWORDS="amd64"

RDEPEND="app-crypt/libsecret
	dev-lang/python
	dev-libs/libgcrypt
	dev-libs/nss
	gnome-base/gconf
	media-libs/alsa-lib
	sys-libs/glibc
	sys-libs/libcap
	virtual/libudev
	x11-base/xorg-x11
	x11-libs/gtk+
	x11-libs/libXtst
	x11-libs/libnotify
	x11-misc/xdg-utils"
DEPEND="net-libs/nodejs"

src_compile () {
	    npm install || die "npm install failed"
	    npm run buildAppImage || "npm run buildAppImage failed"
}
src_install () {
	    mv dist/app/Min-${PV}.AppImage dist/app/min
	    doins /dist/app/min
}
```

**Comment by BlueManCZ** [comment]
Yes, it's quite tricky to write ebuilds for packages based on node.js. You can't use `npm install`, because portage requires all dependencies downloaded and checksummed in the manifest before installation even begins. That would require all packages from `node_modules` to be listed in SRC_URI and then manually placed in the `node_modules` directory and believe me, you don't want to do that. Maybe there are some tools that would do this work for you, but that ebuild would be a nightmare to look at and maintain. That's the reason I just simply extract .deb file in my previously posted ebuild.

But you can try to reach out for help on the official https://forums.gentoo.org/. You will probably get more information there.
